test/e2e/aws_test.go

```diff
@@ -45,101 +47,285 @@ import (
+				deleteCluster(ns2, clName2)
+			})
+		})
+
+		Context("in same namespace", func() {
+			var ns, clName1, clName2 string
+			It("should create first cluster", func() {
+				ns = setup.namespace
+				clName1 = setup.clusterName
+				By("Creating first cluster with single control plane")
+				makeSingleControlPlaneCluster(setup)
+
+				By("Creating the MachineDeployment")
+				createMachineDeployment(setup)
+			})
+			It("should create second cluster in the same namespace", func() {
+				setup.namespace = ns
+				clName2 = setup.clusterName
+				By("Creating second cluster with single control plane")
+				makeSingleControlPlaneCluster(setup)
+
+				By("Creating the MachineDeployment")
+				createMachineDeployment(setup)
+			})
+
+			It("should delete both clusters", func() {
+				By("Deleting the Clusters")
+				deleteCluster(ns, clName1)
+				deleteCluster(ns, clName2)
+			})
+		})
+	})
+
+	Describe("Delete machine lifecycle", func() {
+		It("It should reconcile the deleted machine", func() {
+			By("Creating a workload cluster with single control plane")
+			makeSingleControlPlaneCluster(setup)
+
+			By("Creating the MachineDeployment")
+			createMachineDeployment(setup)
+
+			By("Deleting a worker node machine")
+			deleteMachine(setup.namespace, setup.machineDeploymentName)
+			time.Sleep(10 * time.Second)
+
+			waitForMachineDeploymentRunning(setup.namespace, setup.machineDeploymentName)
+
+			By("Deleting the Cluster")
+			deleteCluster(setup.namespace, setup.clusterName)
+		})
+	})
+
+	Describe("Workload cluster in multiple AZs", func() {
+		It("It should be creatable and deletable", func() {
+			By("Creating a workload cluster with single control plane")
+			setup.multipleAZ = true
+			makeSingleControlPlaneCluster(setup)
+
+			By("Creating the second Control Plane Machine in second AZ")
+			awsMachineName := setup.cpAWSMachinePrefix + "-1"
+			bootstrapConfigName := setup.cpBootstrapConfigPrefix + "-1"
+			machineName := setup.cpMachinePrefix + "-1"
+			setup.availabilityZone = availabilityZones[1]
+			createAdditionalControlPlaneMachine(setup, machineName, awsMachineName, bootstrapConfigName)
+
+			By("Creating the MachineDeployment in second AZ")
+			setup.availabilityZone = availabilityZones[1]
+			createMachineDeployment(setup)
+
+			setup.subnetId = getSubnetId(privateCIDR)
+			By("Checking if the nodes are deployed in second AZ")
+			verfiyInstancesInSubnet(setup.initialReplicas+1, setup.subnetId)
+
+			By("Creating the third Control Plane Machine using SubnetID in second AZ")
+			awsMachineName = setup.cpAWSMachinePrefix + "-2"
+			bootstrapConfigName = setup.cpBootstrapConfigPrefix + "-2"
+			machineName = setup.cpMachinePrefix + "-2"
+			createAdditionalControlPlaneMachine(setup, machineName, awsMachineName, bootstrapConfigName)
+
+			By("Checking if the node is deployed in second AZ")
+			verfiyInstancesInSubnet(setup.initialReplicas+2, setup.subnetId)
+
+			By("Creating the MachineDeployment using SubnetID in second AZ")
```

Comment:
Should this MachineDeployment be in a AZ that is different than the first MachineDeployment created?